clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1142,10 +1142,15 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
       DepFile = "-";
     } else if (ArgMD->getOption().matches(options::OPT_MMD) &&
                Args.hasArg(options::OPT_fintelfpga) &&
-               !Args.hasArg(options::OPT_c)) {
+               JA.isDeviceOffloading(Action::OFK_SYCL)) {
       // When generating dependency files for FPGA AOT, the output files will
-      // always be named after the source file.
-      DepFile = Args.MakeArgString(Twine(getBaseInputStem(Args, Inputs)) + ".d");
+      // always be temporary.  Generate here, which will be used for the aoc
+      // call as well as bundling.
```

Comment:
```suggestion
      // Generate dependency files as temporary. These will be used for
      // the aoc call/bundled during fat object creation.
```
(GH suggestions block me from doing this normally)

Reply:
Done.